### PR TITLE
Added call password to joining a call

### PR DESCRIPTION
**Pull request by jbevan4**
# What
Added a call password to secure calls
# Why
Unsecured calls could of been hijacked
# Screenshots
N/A
# Notes
N/A

### CODE REVIEW

File: pages/visitors/[id]/start.js [hunk 1]
```diff
@@ -49,4 +54,17 @@ const Start = () => {
   );
 };
 
+export const getServerSideProps = propsWithContainer(
```
Comment (danielburnley) [inline pages/visitors/[id]/start.js:57]: Should probably add some tests for this :eyes:

File: pages/visitors/[id]/start.js [hunk 2]
```diff
@@ -49,4 +54,17 @@ const Start = () => {
...
+  async ({ query, container }) => {
+    const { id, callPassword } = query;
+    const callId = id;
+    const queryPassword = callPassword ? callPassword : "";
+    const { scheduledCall, error } = await retrieveVisitByCallId(container)(
+      callId
+    );
+    const dbPassword = scheduledCall.callPassword;
+    return { props: { callId, error, queryPassword, dbPassword } };
+  }
```
Comment (antonyoneill) [inline pages/visitors/[id]/start.js:67]: Technically this would provide the end user with the real password if they looked in the page source

Comment (antonyoneill) [reply pages/visitors/[id]/start.js:67]: I would rather you redirect the user immediately if the password is incorrect. Ie do the check in this function (which runs server side) and if they fail then redirect them to `/error` similar to 
https://github.com/madetech/nhs-virtual-visit/blob/b9c428d21cf0be5cada288211c3315244dfc019a/pages/index.js#L6-L9

File: pages/visitors/[id]/start.js (hunk 1 above)
Comment (jbevan4) [reply pages/visitors/[id]/start.js:57]: Added more tests :D

File: pages/visitors/[id]/start.js (hunk 2 above)
Comment (jbevan4) [reply pages/visitors/[id]/start.js:73]: Copied this pattern 👍

**Pull request merged**